Source/Fuse.Controls.Native/Android/SystemScreenConfigAndroidImpl.uno

```diff
@@ -0,0 +1,104 @@
+using Fuse.Controls.Native;
+
+namespace Fuse.Controls.Native
+{
+	extern(Android) internal class SystemScreenConfigAndroidImpl
+	{
+		private static SystemUiVisibility.Flag _wantedFlags = SystemUiVisibility.Flag.None;
+		private static SystemScreenConfig config;
+
+		public static void Init(SystemScreenConfig config)
+		{
+			SystemScreenConfigAndroidImpl.config = config;
+			SystemUiVisibility.VisibilityChanged += VisibilityChanged;
+		}
+
+		public static void DeInit()
+		{
+			SystemUiVisibility.VisibilityChanged -= VisibilityChanged;
+			SystemScreenConfigAndroidImpl.config = null;
+		}
+
+		public static void TimerDone()
+		{
+			SystemUiVisibility.Flags = _wantedFlags;
+		}
+
+		public static void SetShowState(SystemScreenConfig.Visibility state)
+		{
+			switch(state)
+			{
+				case SystemScreenConfig.Visibility.None:
+					SystemUiVisibility.Flags = SystemUiVisibility.Flag.Fullscreen | SystemUiVisibility.Flag.HideNavigation;
+					_wantedFlags = SystemUiVisibility.Flag.Fullscreen | SystemUiVisibility.Flag.HideNavigation;
+				break;
+				case SystemScreenConfig.Visibility.Minimal:
+					SystemUiVisibility.Flags = SystemUiVisibility.Flag.HideNavigation;
+					_wantedFlags = SystemUiVisibility.Flag.HideNavigation;
+				break;
+				case SystemScreenConfig.Visibility.Full:
+					SystemUiVisibility.Flags = SystemUiVisibility.Flag.None;
+					_wantedFlags = SystemUiVisibility.Flag.None;
+				break;
+			}
+		}
+
+		public static void SetStatusState(bool val)
+		{
+			if(val)
+			{
+				_wantedFlags = _wantedFlags ^ SystemUiVisibility.Flag.Fullscreen;
+				SystemUiVisibility.Flags = SystemUiVisibility.Flags ^ SystemUiVisibility.Flag.Fullscreen;
```

Comment:
xoring flags like that seems pretty fragile; it just flips the SystemUiVisibility.Flag.Fullscreen-bit regardless of what it used to be.

Wouldn't it be better to explicitly clear the flag instead, like so?
```uno
_wantedFlags &= ~SystemUiVisibility.Flag.Fullscreen;
SystemUiVisibility.Flags &= ~SystemUiVisibility.Flag.Fullscreen;
```

Also, it seems strange to track _wantedFlags separately from SystemUiVisibility.Flags... So maybe even this?
```uno
_wantedFlags &= ~SystemUiVisibility.Flag.Fullscreen;
SystemUiVisibility.Flags = _wantedFlags;
```

This seems like it could also get us rid of `getVisibilityFlags`, no?